content/pages/index.js

```diff
@@ -23,18 +23,42 @@ import { Demo } from '../../src/components/common/Demo';
 import schema from './assets/schema.json';
 import uischema from './assets/uischema.json';
 
-const currentVersion = process.env.CURRENTVERSION ?? corePackageJson.version;
-const nextVersion = process.env.NEXTVERSION;
-const nextVersionText =
-  nextVersion && nextVersion !== currentVersion
-    ? `@next: ${nextVersion}`
-    : '';
-
 const data = { firstName: 'Max', lastName: 'Power' };
 
 function Home() {
   const context = useDocusaurusContext();
   const {siteConfig = {}} = context;
+
+  const [currentVersion, setCurrentVersion] = useState(null);
+  const [nextVersion, setNextVersion] = useState(null);
+
+  useEffect(() => {
```

Comment:
As this is implemented as a React `useEffect` hook it will only start fetching the data after the initial rendering, always leading to a flickering UI. Especially so as the `next` is only shown conditionally.

It would be nice if we could
 - do the request outside of the component, so that it's immediately executed even before any React rendering
 - In the render code check whether the data is already there, if yes use it directly. If not we can wait on a promise and then rerender once the data is there. This can be done in a `useEffect`.
 - As long as the data is not there yet and the request did not fail, do not render any version number in the UI but render empty space there. Then replace the empty space with the retrieved / stored version number once it's  completed failed. The UI should not jump then. This also means we now always reserves space for "next".